lib/api/objectDelete.js

```diff
@@ -99,6 +99,21 @@ function objectDelete(authInfo, request, log, cb) {
                     return next(null, bucketMD, objectMD, options);
                 });
         },
+        function validateHeaders(bucketMD, objectMD, options, next) {
+            if (objectMD) {
+                const lastModified = objectMD['last-modified'];
+                const { modifiedSinceRes, unmodifiedSinceRes } =
+                    checkDateModifiedHeaders(request.headers, lastModified);
+                const err = modifiedSinceRes.error || unmodifiedSinceRes.error;
+                if (err) {
+                    return process.nextTick(() => next(err, bucketMD));
+                }
+                return process.nextTick(() =>
```

Comment:
curious, why `process.nextTick`?

Reply:
cos everything in the function is synchronous, so we gotta mimic async :)

Reply:
cool beans ty for the explanation :)